### PR TITLE
sway-contrib: unstable-2023-06-30 -> 0-unstable-2024-01-20

### PR DESCRIPTION
**Pull request by tr3foil**
## Description of changes

Upstream doesn't have a changelog that I could see.

More scripts have been added to the repo (which as of now are not packaged), along with updates to the existing ones.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

CC: @evils 

Planning on doing more testing tomorrow when I'm back at the desktop. :+1:

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/sway-contrib/default.nix [hunk 1]
```diff
@@ -15,12 +15,12 @@
 }:
 
 let
-  version = "unstable-2023-06-30";
+  version = "unstable-2024-01-20";
```
Comment (kirillrdy) [inline pkgs/applications/misc/sway-contrib/default.nix:18]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

correct version should be `0-unstable-2024-01-20`

Comment (kirillrdy) [reply pkgs/applications/misc/sway-contrib/default.nix]: no worries, can you also update PR title ?

**Comment by pyrotelekinetic** [comment]
Should I move the package to `pkgs/by-name/`? The [manual migration guidlines](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#manual-migration-guidelines) say it should be lightly encouraged while the code is being worked on anyway.

In [`all-packages.nix`](https://github.com/NixOS/nixpkgs/blob/75c29d4ecea2234955183f0d7e843c629fd01459/pkgs/top-level/all-packages.nix#L32597) the package is called with `recurseIntoAtttrs (callPackages ../applications/misc/sway-contrib)`, so it looks like the entry will have to stay even if we moved the package to by-name.

**Comment by kirillrdy** [comment]
@pyrotelekinetic up to you. There is automated migration being worked on, and wording in the migration guidelines is trying to prevent people flooding with PR's that just move files around

**Comment by pyrotelekinetic** [comment]
I think I won't move to by-name then, since the entry in `all-packages.nix` would still be needed anyhow.

**Comment by evils** [comment]
hmm, unfortunately the newer grimshot in this does not highlight the selected area...

**Comment by evils** [comment]
their new `anything` option is quite nice though, and seems to work as an adequate replacement for `area`

**Comment by pyrotelekinetic** [comment]
I imagine this is a regression in `slurp`, I'll play around with it today.

**Comment by pyrotelekinetic** [comment]
@evils I can't reproduce this; `grimshot copy area` looks identical for me on nixos-unstable and on this pr. 

nixos-unstable (e92039b55bcd58469325ded85d4f58dd5a4eaf58):
nixos-unstable

This pr:
update-sway-contrib

How exactly are you testing?

**Comment by evils** [comment]
weird, now it works here as well
this is in a nixpkgs-review shell

i've since rebuilt and rebooted my system, so i'm guessing this was a compatibility issue with an older sway?

**Comment by pyrotelekinetic** [comment]
@evils Would you mind doing a PR review? I think that's the last thing preventing this being merged.

**Review by evils — APPROVED** [review]
grimshot works here
assuming this also fixes the license and platforms for the sub-packages

**Comment by pyrotelekinetic** [comment]
Force pushed for a reword - I had written `Refactor` instead of `refactor`.

I had to fix it. I would never get over it if I didn't.

**Review by evils — APPROVED** [review]
i approve of your commitment to commit message formatting :)

**Comment by teto** [comment]
Result of `nixpkgs-review pr 283388` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>sway-contrib.grimshot</li>
    <li>sway-contrib.grimshot.man</li>
    <li>sway-contrib.inactive-windows-transparency</li>
  </ul>
</details>

**Pull request merged**